drivers/serial/uart_nrfx_uarte.c

```diff
@@ -877,6 +931,19 @@ static int uarte_nrfx_rx_enable(const struct device *dev, uint8_t *buf,
 		return -EBUSY;
 	}
 
+#ifdef CONFIG_HAS_NORDIC_DMM
```

Comment:
this could should compile fine for legacy targets as well - `dmm_buffer_in_prepare` is mocked for them: https://github.com/zephyrproject-rtos/zephyr/blob/98e0e65069cf62297c4d03e0ad5b543374f5b735/soc/nordic/common/dmm.h#L134

Reply:
im ifdefing because for dmm case there are additional fields in the data and config structure to hold user buffer (as it is different one than one used for DMA) and `mem_reg`. We could keep this for non-dmm cases and that would work but then its redundant so i preferred to ifdef it.